tests/helper_test.py

```diff
@@ -0,0 +1,235 @@
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test suite for alot.helper module."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import datetime
+import random
+import unittest
+
+import mock
+
+from alot import helper
+
+
+class Test_split_commandline(unittest.TestCase):
+
+    def _test(self, base, expected):
+        """Shared helper to reduce some boilerplate."""
+        actual = helper.split_commandline(base)
+        self.assertListEqual(actual, expected)
+
+    def test_simple(self):
+        base = 'echo "foo";sleep 1'
+        expected = ['echo "foo"', 'sleep 1']
+        self._test(base, expected)
+
+    def test_single(self):
+        base = 'echo "foo bar"'
+        expected = [base]
+        self._test(base, expected)
+
+    def test_unicode(self):
+        base = u'echo "foo";sleep 1'
+        expected = ['echo "foo"', 'sleep 1']
+        self._test(base, expected)
+
+    def test_escapes(self):
+        base = 'echo "foo \"bar\"";'
+        expected = ['echo "foo \"bar\""']
+        self._test(base, expected)
+
+
+class Test_split_commandstring(unittest.TestCase):
+
+    def _test(self, base, expected):
+        """Shared helper to reduce some boilerplate."""
+        actual = helper.split_commandstring(base)
+        self.assertListEqual(actual, expected)
+
+    def test_bytes(self):
+        base = b'echo "foo bar"'
+        expected = [b'echo', b'foo bar']
+        self._test(base, expected)
+
+    def test_unicode(self):
+        base = u'echo "foo €"'
+        expected = [b'echo', u'foo €'.encode('utf-8')]
+        self._test(base, expected)
+
+
+class Test_string_sanitize(unittest.TestCase):
+
+    def test_tabs(self):
+        base = 'foo\tbar\noink\n'
+        expected = 'foo' + ' ' * 5 + 'bar\noink\n'
+        actual = helper.string_sanitize(base)
+        self.assertEqual(actual, expected)
+
+
+class Test_string_decode(unittest.TestCase):
+
+    def _test(self, base, expected, encoding='ascii'):
+        actual = helper.string_decode(base, encoding)
+        self.assertEqual(actual, expected)
+
+    def test_ascii_bytes(self):
+        base = u'test'.decode('ascii')
+        expected = u'test'
+        self._test(base, expected)
+
+    def test_utf8_bytes(self):
+        base = u'test'.decode('utf-8')
+        expected = u'test'
+        self._test(base, expected, 'utf-8')
+
+    def test_unicode(self):
+        base = u'test'
+        expected = u'test'
+        self._test(base, expected)
+
+
+class Test_pretty_datetime(unittest.TestCase):
+
+    # TODO: Currently these tests use the ampm format based on whether or not
+    # the testing machine's locale sets them. To be really good mock should be
+    # used to change the locale between an am/pm locale and a 24 hour locale
+    # and test both scenarios.
+
+    __patchers = []
```

Comment:
It's a private class variable to be used by the class level fixtures (`setUpClass` and `tearDownClass`), and shouldn't be touched elsewhere. This is completely okay via PEP8, it invokes python's name mangling (in CPython this will be changed to `__Test_pretty_datetime_patchers`, which is the closest thing python has to a private attribute/method.

It's down near the bottom of [this section of PEP8](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles)